### DISCUSSION TITLE
Tutorial 00, objective `error_metric` vs `error_squared_norm`?

### DISCUSSION THREAD

**Discussion by DanielTakeshi** [facebookresearch/theseus#531] [Q&A] [answered]
Has there been a recent change that deprecated `error_metric()` in favor of `error_squared_norm()`? I'm running tutorial 00 for Theseus, from today (as of `main` branch) and am running into this issue when I try and run the "Objective" category.

![Screenshot from 2023-06-12 12-04-18](https://github.com/facebookresearch/theseus/assets/5532096/09d0090e-106e-4d46-be41-036825b5bb42)

For ease of copy and paste, the error text is below:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[6], line 35
     33 print(f"Error term 2: unweighted: {cf2.error()} weighted: {cf2.weighted_error()}")
     34 # Objective value should be: (error1)^2 + (error2)^2 
---> 35 print(f"Objective value: {objective.error_metric()}")

AttributeError: 'Objective' object has no attribute 'error_metric'
```

However, switching it to use `error_squared_norm()` seems to be fine:

![Screenshot from 2023-06-12 12-07-05](https://github.com/facebookresearch/theseus/assets/5532096/bfca1c35-9d1a-4b38-9a97-7ca6cae08c2b)

This value of 4.250 also matches the one that I now see on the public GitHub:

https://github.com/facebookresearch/theseus/blob/main/tutorials/00_introduction.ipynb

The relevant part is located here and uses `error_metric()`:

![Screenshot from 2023-06-12 12-12-17](https://github.com/facebookresearch/theseus/assets/5532096/41e31dea-c21c-4ada-a8fa-7c25f6cabbb2)

Has there been a recent change which deprecated some of the APIs? If so, might be worth updating the notebook examples.

**Comment by luisenp** [comment, marked as answer]
Ah, the `error_metric` method was introduced recently, after version 0.1.4. The tutorials are consistent with the latest source code, but I missed that updating tutorials with this new change would break them for older versions of `theseus`. Sorry about this. 

Switching to `error_squared_norm()` is the correct thing to do if you are on an older version.

**Reply by DanielTakeshi** [reply]
Got it, thanks! I think PyPI still uses 0.1.4 https://pypi.org/project/theseus-ai/ but presumably that will get updated sometime in the future.

**Reply by luisenp** [reply]
Yes. Should be updated to 0.2.0 by the end of this month.
